### PR TITLE
Handle parents provided by StateObject references

### PR DESCRIPTION
**Pull request by paweld2**
In the case that the 'parent' property of a state is provided as a reference to a StateObject, the function $$parentState returns a reference to the State object and not the state name. On this case the application crash.

I added a test case for this and a fix on $$parentState.

### CODE REVIEW

File: src/angular-breadcrumb.js [hunk 1]
```diff
@@ -42,8 +42,13 @@ function $Breadcrumb() {
...
         var $$parentState = function(state) {
             // Check if state has explicit parent OR we try guess parent from its name
-            var name = state.parent || (/^(.+)\.[^.]+$/.exec(state.name) || [])[1];
+            var parent = state.parent || (/^(.+)\.[^.]+$/.exec(state.name) || [])[1];
             // If we were able to figure out parent name then get this state
+            var name = parent;
+            if( typeof parent === "object" ) {
+                // if parent is a object reference, then extract the name
+                name = parent.name;
+            }
```
Comment (ncuillery) [inline src/angular-breadcrumb.js]: How about a ternary operator like [this one](https://github.com/ncuillery/angular-breadcrumb/blob/master/src/angular-breadcrumb.js#L87) ?

**Comment by ncuillery** [comment]
Thanks for the PR, clean and tested !

I just leave a note above. Additionally, can you amend the commit and type this message:
`fix($breadcrumb): Handle parents provided by StateObject references` ? I use [this convention](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-git-commit-guidelines) to generate the CHANGELOG.md with `grunt-conventional-changelog`.

**Comment by paweld2** [comment]
Code fixed and message changed.

**Comment by ncuillery** [comment]
Landed, thanks a lot !

**Pull request merged**